network/ocsp.go

```diff
@@ -144,26 +153,35 @@ func queryOCSP(commonName string, clientCert, issuerCert *x509.Certificate, ocsp
 	return ocspResponse, err
 }
 
+type OCSPVerifier interface {
+	// Verify returns number of confirmations or error.
+	// The error is returned only if it is critical, for example:
+	// - the certificate was revoked
+	// - the certificate is not known by OCSP server and we requested tls_ocsp_required == "yes" or "all"
+	// - if we were unable to contact OCSP server(s) but we really need the response, tls_ocsp_required == "all"
+	Verify(chain []*x509.Certificate) (int, error)
+}
```

Comment:
This interface seems to be the same as `CRLVerifier`. Do you plan for them to be different eventually?

In my opinion, certificate verification is something generic: it can be CRL blacklists, interactive checks by OCSP, or even interactive query to the user if Acra were a GUI app. So it makes more sense, I think, to have a generic enough `Verifier` interface with assorted implementations for it. And documentations comments for particular `Verify()` implementation might describe their specifics.

Reply:
Agree. Probably better to rely on `VerifyPeerCertificate` function signature: `func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error`, to provide access to all data from connection which may be used for all our verifiers, and to avoid any adapters for that.